src/renderer/renderer_types.zig

```diff
@@ -14,27 +15,31 @@ pub const MeshLod = struct {
     vertex_count: u32,
 };
 
-pub const BoundingBox = struct {
-    min: [3]f32,
-    max: [3]f32,
-};
+pub const SubMesh = struct {
+    lod_count: u32,
+    lods: [lod_count_max]MeshLod,
 
-pub const BoundingBoxCoordinates = struct {
-    center: [3]f32,
-    radius: f32,
+    bounding_box: BoundingBox,
 };
 
 pub const Mesh = struct {
     vertex_buffer: BufferHandle,
     index_buffer: BufferHandle,
-    num_lods: u32,
-    lods: [max_num_lods]MeshLod,
+
+    sub_mesh_count: u32,
+    sub_meshes: [sub_mesh_count_max]SubMesh,
+
     bounding_box: BoundingBox,
+};
+
+pub const BoundingBox = struct {
+    min: [3]f32,
+    max: [3]f32,
 
-    pub fn calculateBoundingBoxCoordinates(mesh: *const Mesh, z_world: zm.Mat) BoundingBoxCoordinates {
-        var z_bb_min = zm.loadArr3(mesh.bounding_box.min);
+    pub fn calculateBoundingBoxCoordinates(self: *const BoundingBox, z_world: zm.Mat) BoundingBoxCoordinates {
+        var z_bb_min = zm.loadArr3(self.min);
         z_bb_min[3] = 1.0;
```

Comment:
How come we ignore z?

Reply:
No, I'm setting the W component of the vector to 1, so they get treated as positions (and not directions) during the world-space transformation :)

Reply:
This comment chain never happened.

Reply:
We can move the repo, squash all history and destroy all evidence!